### PR TITLE
Add more null annotations to XML processing classes

### PR DESCRIPTION
**Pull request by wborn**
This adds null annotations to many XML processing classes and a few others.

### CODE REVIEW

**Review by kaikreuzer — APPROVED** [review]
Thanks, looks good from what I can see.
One final question, though, as you have added "API breaking" to the PR: I assume this concerns only APIs that are not being used by any (official) add-ons and thus it does not really have an impact. Is this correct?

**Comment by wborn** [comment]
Yes that's right. The add-ons are not impacted by this. I added the label because I removed the `ConverterAssertion` utility class. I found that only the `assertEndOfType` method in that class was used by the `ConfigDescriptionConverter` . So it seemed like a nice simplification to just move those 3 lines to where they were actually used.

**Comment by kaikreuzer** [comment]
Thanks!

File: bundles/org.openhab.core.thing.xml/src/main/java/org/openhab/core/thing/xml/internal/ChannelTypeConverter.java [hunk 1]
```diff
@@ -70,29 +74,30 @@ private boolean readBoolean(Map<String, String> attributes, String attributeName
     }
 
     private String readItemType(NodeIterator nodeIterator) throws ConversionException {
-        return (String) nodeIterator.nextValue("item-type", false);
+        return requireNonEmpty((String) nodeIterator.nextValue("item-type", false),
+                "ChannelType 'itemType' must not be null or empty.");
```
Comment (J-N-K) [inline bundles/org.openhab.core.thing.xml/src/main/java/org/openhab/core/thing/xml/internal/ChannelTypeConverter.java:78]: @wborn I think this change causes #2831. For trigger-channels an empty `itemType` is ok.

Comment (J-N-K) [inline bundles/org.openhab.core.thing.xml/src/main/java/org/openhab/core/thing/xml/internal/ChannelTypeConverter.java:78]: @wborn I think this change causes #2831. For trigger-channels an empty `itemType` is ok.

**Pull request merged**